### PR TITLE
X86 fix

### PR DESCRIPTION
**Pull request by serpilliere**
Multiple fixes in `x86` semantic/architecture.

### CODE REVIEW

File: miasm2/arch/x86/regs.py [hunk 1]
```diff
@@ -342,6 +342,22 @@
 float_st6 = ExprId("float_st6", 64)
 float_st7 = ExprId("float_st7", 64)
 
+float_replace = {ExprId('ST', 64): float_st0,
+                 fltregs32_expr[0]: float_st0,
```
Comment (commial) [inline miasm2/arch/x86/regs.py]: By declaring `float_list` before, you can use:
```Python
float_replace = {fltregs32_expr[i]: float_list[i] for i in xrange(8)}
float_replace[ExprId('ST', 64)] = float_st0
```

File: miasm2/arch/x86/regs.py [hunk 2]
```diff
@@ -342,6 +342,22 @@
 float_st6 = ExprId("float_st6", 64)
 float_st7 = ExprId("float_st7", 64)
 
+float_replace = {ExprId('ST', 64): float_st0,
```
Comment (commial) [inline miasm2/arch/x86/regs.py]: `ExprId('ST', 64)` is already declared at line 88, may a variable can avoid future error.

File: miasm2/expression/expression.py [hunk 3]
```diff
@@ -760,13 +761,20 @@ def __init__(self, op, *args):
                           'int_16_to_double', 'int_32_to_double',
                           'int_64_to_double', 'int_80_to_double']:
             sz = 64
-        elif self._op in ['double_to_mem_16', 'double_to_int_16', 'double_trunc_to_int_16']:
+        elif self._op in ['double_to_mem_16', 'double_to_int_16',
```
Comment (commial) [inline miasm2/expression/expression.py]: For `double_to_mem_`, `double_to_int_`, `double_trunc_to_int_` it seems that the size is embedded in the `ExprOp` name.
Using a ```if ... startswith``` and getting the size at the end  may allow to automatically supporting every size for these instructions (if this is a wanted behavior), avoiding copying the same code for each new size to support.

Comment (serpilliere) [inline miasm2/expression/expression.py]: I prefer (for the moment) to keep a white list for such sanity checks

**Pull request merged**